server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1255,9 +1301,68 @@ public void deleteConsumer(
                 returnCerts.add(cert);
             }
         }
+        // we want to insert the content access cert to this list
+        if (!StringUtils.isEmpty(consumer.getOwner().getContentAccessMode())) {
+            try {
+                Certificate cert = contentAccessCertService.getCertificate(consumer);
+                if (cert != null) {
+                    returnCerts.add(cert);
+                }
+            }
+            catch (IOException ioe) {
+                throw new BadRequestException(i18n.tr("Cannot retrieve content access certificate"), ioe);
+            }
+            catch (GeneralSecurityException gse) {
+                throw new BadRequestException(i18n.tr("Cannot retrieve content access certificate"), gse);
+            }
+        }
         return returnCerts;
     }
 
+    @ApiOperation(notes = "Retrieves the body of the Content Access Certificate for the Consumer",
+        value = "getContentAccessBody")
+    @ApiResponses({ @ApiResponse(code = 404, message = ""), @ApiResponse(code = 304, message = "") })
+    @GET
+    @Path("{consumer_uuid}/content_access")
+    @Produces(MediaType.APPLICATION_JSON)
+    public ContentAccessListing getContentAccessBody(
+        @PathParam("consumer_uuid") @Verify(Consumer.class) String consumerUuid,
+        @QueryParam("if-modified-since") Date since) {
+
+        log.debug("Getting content access certificate for consumer: {}", consumerUuid);
+        Consumer consumer = consumerCurator.verifyAndLookupConsumer(consumerUuid);
+        if (StringUtils.isEmpty(consumer.getOwner().getContentAccessMode())) {
+            throw new BadRequestException(i18n.tr("No content access mode assigned"));
+        }
+        if (!contentAccessCertService.hasCertChangedSince(consumer, since)) {
+            throw new NotModifiedException("Not modified since date supplied.");
```

Comment:
I'm not 100% sure we want to throw an exception here since this isn't really an exceptional case.  In fact, we expect it to happen frequently.  Instead we can just use RestEasy's response builder to build a response with a 304.  Using exceptions for control flow can result in poor performance and some other issues as described by [item 57 in Effective Java](https://www.safaribooksonline.com/library/view/effective-java-2nd/9780137150021/ch09.html)